src/ngraph/interval.hpp

```diff
@@ -0,0 +1,120 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+
+#include "ngraph/ngraph_visibility.hpp"
+
+namespace ngraph
+{
+    /// \brief Interval arithmetic
+    ///
+    /// An interval is the set of integers from m_min_val through m_max_val.
+    /// The value s_max acts like infinity. The
+    /// addition, subtraction, or multiplication of intervals is the smallest interval
+    /// containing the sums, differences, or products of elements of the two intervals. An empty
+    /// interval is canonicalized to [s_max, s_max].
+    class NGRAPH_API Interval
+    {
+    public:
+        using value_type = std::int64_t;
```

Comment:
This is the second definition of value_type, the first being in dimension. This should be defined in a single location so they don't get out of sync.

Reply:
No need for them to be in sync.